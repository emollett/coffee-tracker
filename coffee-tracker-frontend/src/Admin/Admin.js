import React, { Component } from "react";
import "../App.css";
import Input from "../Input/Input";
import Edit from "../edit.js";
import List from "../List/List";
import { Tab, Tabs, TabList, TabPanel } from "react-tabs";
import { withRouter} from "react-router-dom";


class Admin extends Component {

	render() {

		return (
			<div className="bigPadding">

				<Tabs>
					<TabList>
						<Tab><h3>Add a coffee</h3></Tab>
						<Tab><h3>Edit an entry</h3></Tab>
					</TabList>

					<TabPanel>
						<Input data={this.props.data}/>
					</TabPanel>

					<TabPanel>
						<Edit data={this.props.data}/>
					</TabPanel>
				</Tabs>

				<List data={this.props.data}/>

			</div>
		);
	}
}

export default withRouter(Admin);
