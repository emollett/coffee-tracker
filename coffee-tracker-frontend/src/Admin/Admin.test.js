import React from "react";
import ReactDOM from "react-dom";
import Admin from "./Admin";
import testData from "../testdata.json";
import renderer from "react-test-renderer";
import { MemoryRouter as Router } from "react-router-dom";

it("renders without crashing", () => {
	const div = document.createElement("div");
	ReactDOM.render(<Router><Admin data={testData}/></Router>, div);
	ReactDOM.unmountComponentAtNode(div);
});

it("renders correctly", () => {
	const tree = renderer
		.create(<Router><Admin data={testData}/></Router>)
		.toJSON();
	expect(tree).toMatchSnapshot();
});