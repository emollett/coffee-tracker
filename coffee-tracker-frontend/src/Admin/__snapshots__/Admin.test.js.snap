// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="bigPadding"
>
  <div
    className="react-tabs"
    data-tabs={true}
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <ul
      className="react-tabs__tab-list"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-5"
        aria-disabled="false"
        aria-selected="true"
        className="react-tabs__tab react-tabs__tab--selected"
        id="react-tabs-4"
        role="tab"
        tabIndex="0"
      >
        <h3>
          Add a coffee
        </h3>
      </li>
      <li
        aria-controls="react-tabs-7"
        aria-disabled="false"
        aria-selected="false"
        className="react-tabs__tab"
        id="react-tabs-6"
        role="tab"
        tabIndex={null}
      >
        <h3>
          Edit an entry
        </h3>
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-4"
      className="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-5"
      role="tabpanel"
    >
      <form
        className="bigPadding"
        id="new-coffee-form"
      >
        <div
          className="smallPadding"
        >
          <label
            htmlFor="coffee"
          >
            Add a coffee to the database
          </label>
          <br />
          <input
            id="coffee"
            onChange={[Function]}
            required={true}
            type="text"
          />
        </div>
        <div
          className="smallPadding"
        >
          <label
            htmlFor="name"
          >
            Who bought the coffee?
          </label>
          <br />
          <input
            autoComplete="given-name"
            id="name"
            onChange={[Function]}
            required={true}
            type="text"
          />
        </div>
        <div
          className="smallPadding"
        >
          <label
            htmlFor="datePurchased"
          >
            What date was it purchased?
          </label>
          <br />
          <input
            id="datePurchased"
            onChange={[Function]}
            type="date"
          />
        </div>
        <div
          className="smallPadding"
        >
          <label
            htmlFor="dateOpened"
          >
            What date was it opened?
          </label>
          <br />
          <input
            id="dateOpened"
            onChange={[Function]}
            type="date"
          />
        </div>
        <div
          className="smallPadding"
        >
          <label
            htmlFor="message"
          >
            Add a note about the coffee
          </label>
          <br />
          <input
            id="message"
            onChange={[Function]}
            type="text"
          />
        </div>
        <div
          className="smallPadding"
        >
          <button
            data-testid="submitCoffeeTest"
            onClick={[Function]}
          >
            <h4>
              Add to database
            </h4>
          </button>
        </div>
      </form>
    </div>
    <div
      aria-labelledby="react-tabs-6"
      className="react-tabs__tab-panel"
      id="react-tabs-7"
      role="tabpanel"
    />
  </div>
  <div
    className="bigPadding"
  >
    <div
      className="smallPadding"
    >
      <table
        className="table"
      >
        <thead>
          <tr>
            <th>
              ID
            </th>
            <th>
              Name
            </th>
            <th>
              Date Purchased
            </th>
            <th>
              Date Opened
            </th>
            <th>
              Coffee
            </th>
            <th>
              Note
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              1
            </td>
            <td>
              Alice
            </td>
            <td>
              <time
                dateTime={"2018-05-13T07:01:41.522Z"}
              >
                13/05/2018
              </time>
            </td>
            <td>
              <time
                dateTime={"2018-05-13T07:01:41.522Z"}
              >
                13/05/2018
              </time>
            </td>
            <td>
              Tesco
            </td>
            <td>
              Hello
            </td>
          </tr>
          <tr>
            <td>
              2
            </td>
            <td>
              Alice
            </td>
            <td>
              <time
                dateTime={"2018-07-13T07:01:41.522Z"}
              >
                13/07/2018
              </time>
            </td>
            <td>
              <time
                dateTime={"2018-07-13T07:01:41.522Z"}
              >
                13/07/2018
              </time>
            </td>
            <td>
              Aldi
            </td>
            <td>
              Hello
            </td>
          </tr>
          <tr>
            <td>
              3
            </td>
            <td>
              Betty
            </td>
            <td>
              <time
                dateTime={"2018-07-13T07:01:41.522Z"}
              >
                13/07/2018
              </time>
            </td>
            <td>
              <time
                dateTime={"2018-07-13T07:01:41.522Z"}
              >
                13/07/2018
              </time>
            </td>
            <td>
              Tesco
            </td>
            <td>
              Hello
            </td>
          </tr>
          <tr>
            <td>
              4
            </td>
            <td>
              Cait
            </td>
            <td>
              <time
                dateTime={"2018-06-13T07:01:41.522Z"}
              >
                13/06/2018
              </time>
            </td>
            <td>
              <time
                dateTime={"2018-06-13T07:01:41.522Z"}
              >
                13/06/2018
              </time>
            </td>
            <td>
              Tesco
            </td>
            <td>
              Hello
            </td>
          </tr>
          <tr>
            <td>
              5
            </td>
            <td>
              Cait
            </td>
            <td>
              <time
                dateTime={"2018-06-13T07:01:41.522Z"}
              >
                13/06/2018
              </time>
            </td>
            <td>
              <time
                dateTime={"2018-06-13T07:01:41.522Z"}
              >
                13/06/2018
              </time>
            </td>
            <td>
              Tesco
            </td>
            <td>
              Hello
            </td>
          </tr>
          <tr>
            <td>
              6
            </td>
            <td>
              Alice
            </td>
            <td>
              <time
                dateTime={"2019-07-13T07:01:41.522Z"}
              >
                13/07/2019
              </time>
            </td>
            <td>
              <time
                dateTime={"2019-07-13T07:01:41.522Z"}
              >
                13/07/2019
              </time>
            </td>
            <td>
              Aldi
            </td>
            <td>
              Hello
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
